ovirt_imageio/client/_ui.py

```diff
@@ -8,10 +8,54 @@
 from .. _internal import util
 
 
+DEFAULT_WIDTH = 79
+
+
+class OutputFormat:
+
+    def __init__(self, now, size=None, width=DEFAULT_WIDTH):
+        """
+        Arguments:
+            now (callable): callable returning current time for testing.
+            size (int): total number of bytes. If size is unknown when
+                creating, progress value is not displayed. The size can be set
+                later to enable progress display.
+            width (int): width of progress bar in characters (default 79)
+        """
+        self._now = now
+        self._start = self._now()
+        self.size = size
+        # TODO: use current terminal width instead.
+        self._width = width
+
+    def draw(self, value, transferred, phase=None, last=False):
+        raise NotImplementedError
+
+
+class TextFormat(OutputFormat):
```

Comment:
In the next patch you call the format "human", why not use the same term in the actual class name?

Or if text is the better term, why not call it "text" in the options?

I think text is a better term, since "human" is not a format, and text is considered "the universal interface":
https://scale.com/blog/text-universal-interface

This is also the term we add to clusteradm here:
https://github.com/open-cluster-management-io/clusteradm/pull/288

Reply:
Good point about qemu, but I think we should look at more popular examples. Here is
minikube version:

```
    -o, --output='text':
	Format to print stdout in. Options include: [text,json]
```